### ISSUE TITLE
[HOLD for payment 2025-01-15] [$125] Add "Go to workspace" as a shortcut in all of the workspace chats, rooms and reports details page. 

### ISSUE THREAD

**Issue by trjExpensify** [Expensify/App#52817]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.64-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @trjExpensify 
**Slack conversation**: [#convert thread](https://expensify.slack.com/archives/C07HPDRELLD/p1731967223721749)

## Action Performed:
1. Create a workspace 
2. Navigate to the #admins room
3. Click the header of the chat

## Expected Result:
_This is a new feature_ 

1. Add a `Go to workspace` row in the bottom position of the details screen of all workspace chats, rooms, threads originating from a parentChat associated with a workspace, expense reports and expense transaction threads. 

<img width="353" alt="image" src="https://github.com/user-attachments/assets/ca9846a8-06a5-4f78-82bf-507eb733142d">

2. When clicked, navigate to the workspace settings page. On small screens that'll be the LHN workspace settings page, on large screens that will be the workspace profile page. 

## Actual Result:
Users are getting a little lost trying to find their workspace settings, and one of the places they're commonly looking is in the details screen of the chats/reports associated with the workspace. As such, we want to provide a shortcut to get to the workspace settings from there to hopefully help. 

## Workaround:
N/A, improvement. 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
In-line above. 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021860062325064165317</li>
        <li>Upwork Job ID: 1860062325064165317</li>
        <li>Last Price Increase: 2024-11-22</li>
<li>Automatic offers: </li>
<ul>
<li>mkzie2 | Contributor | 105028511</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Users are getting a little lost trying to find their workspace settings, and one of the places they're commonly looking is in the details screen of the chats/reports associated with the workspace. As such, we want to provide a shortcut to get to the workspace settings from there to hopefully help.

### What is the root cause of that problem?

This is a new feature

### What changes do you think we should make in order to solve the problem?

1. Create a new const here for `Go to workspace`

https://github.com/Expensify/App/blob/9f83a4118ff0b8080ac8c037b36a26252a12b06f/src/CONST.ts#L3150

2.  Add the `Go to workspace` item here if the `report?.policyID` isn't fake policyID with `action` is a function that we will navigate to the LHN workspace settings page, navigate to the workspace profile page if it's the large screen.

```
items.push({
    key: CONST.REPORT_DETAILS_MENU_ITEM.GO_TO_WORKSPACE,
    translationKey: 'common.goToWorkspace',
    icon: Expensicons.Workspace,
    action: () => {
        if (isSmallScreenWidth) {
            Navigation.navigate(ROUTES.WORKSPACE_INITIAL.getRoute(report?.policyID));
            return;
        }
        Navigation.navigate(ROUTES.WORKSPACE_PROFILE.getRoute(report?.policyID));
    },
    isAnonymousAction: false,
    shouldShowRightIcon: true,
})
```
https://github.com/Expensify/App/blob/9f83a4118ff0b8080ac8c037b36a26252a12b06f/src/pages/ReportDetailsPage.tsx#L520



### What alternative solutions did you explore? (Optional)

**Comment by mananjadhav** [comment]
Proposal from @mkzie2 looks good to me.

🎀 👀 🎀 C+ reviewed.

**Comment by mananjadhav** [comment]
@youssef-lr I am not sure if something else needs to be done. @mkzie2 was assigned the issue but I don't see a GH comment about the assignment + upwork link.

**Comment by youssef-lr** [comment]
yeah it seems like it failed to do so. Let me try again.

**Comment by youssef-lr** [comment]
Hmm it's not working, I'll check the logs and report back

**Comment by youssef-lr** [comment]
Adjusting price as this is a pretty simple feature request.

**Comment by mkzie2** [comment]
@trjExpensify Do we always want to keep `Go to workspace` at the bottom? For example in thread report & transaction thread report:

| Thread  | Transaction thread  |
|---|---|
| ![Screenshot 2024-11-25 at 23 42 28](https://github.com/user-attachments/assets/cb5d6745-ce75-4272-ade0-607c67c7aab9) |  ![Screenshot 2024-11-25 at 23 42 41](https://github.com/user-attachments/assets/75b645d2-35f3-412e-953d-0c836807ebf2) |

**Comment by trjExpensify** [comment]
Good question! @Expensify/design, what do you think? Breaking up the carets looks a bit weird.

**Comment by shawnborton** [comment]
Agree, I think I would keep Delete last.

**Comment by trjExpensify** [comment]
Cool, sounds good to me!

**Comment by mananjadhav** [comment]
Waiting on @mkzie2 to fix the comments raised on the PR.

**Comment by mananjadhav** [comment]
@trjExpensify @youssef-lr Can one of you confirm the behavior [here](https://github.com/Expensify/App/pull/53076#issuecomment-2518051325)?

**Comment by mananjadhav** [comment]
The PR was reverted. Awaiting the revised PR.

**Comment by mkzie2** [comment]
@mananjadhav The [PR](https://github.com/Expensify/App/pull/53898) is ready.

**Comment by trjExpensify** [comment]
👋 checklist time please, @mananjadhav!

**Comment by mananjadhav** [comment]
@trjExpensify There was a regression on the PR, so it was reverted. The follow-up PR is raised, which I'll finish the checklist on.

**Comment by trjExpensify** [comment]
Great, thanks!

**Comment by mananjadhav** [comment]
@mkzie2 Quick bump on the PR comment to fix the lint issues.

**Comment by mananjadhav** [comment]
The PR is reviewed, waiting for @youssef-lr to review and merge.

**Comment by trjExpensify** [comment]
Awaiting a [secondary PR review](https://github.com/Expensify/App/pull/53898#issuecomment-2568325506) from @youssef-lr.

**Comment by trjExpensify** [comment]
👋 checklist time here!

**Comment by mananjadhav** [comment]
Yes. I'll work through this. 

But wanted to quickly raise @trjExpensify @youssef-lr. I was wondering if we could keep the regular pay of $250. The change wasn't like a 1-2 liner change as expected. We did some refactor too. This also had a regression so essentially I am request for $125 payout. Let me know what you folks think?

**Comment by youssef-lr** [comment]
I'm cool with that @mananjadhav

**Comment by trjExpensify** [comment]
Sounds good!

**Comment by mananjadhav** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: Can we treated as a feature request.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1. Go to any workspace expense chat.
2. Press report header.
3. Verify "Go to workspace" menu shows.
4. Press Go to workspace
5. Verify app opens workspace initial settings page on small screens and workspace profile page on large screens
6. Repeat steps 2 - 5 with workspace rooms, workspace's expense reports, workspace's transaction thread reports, thread reports originating from any kind of workspace chats listed above
7. Enable the invoices feature
8. Create a send invoice
9. Delete the workspace
10. Go the report detail of the invoice report and verify that no `Leave` option
11. Open the report detail of the workspace chat and verify that no `Go to workspace` option


Do we agree 👍 or 👎


</details>

**Comment by trjExpensify** [comment]
Payment summary as follows: 

- $125 to @mananjadhav for the C+ review (go ahead and request!)
- $125 to @mkzie2 for the fix (paid!

Regression test issue created [here](https://github.com/Expensify/Expensify/issues/461201). Closing, thanks!

**Comment by garrettmknight** [comment]
$125 approved for @mananjadhav
